File: backend/tiggle-root/tiggle/src/main/java/com/side/tiggle/domain/tag/TagDto.java
package com.side.tiggle.domain.tag;

import com.side.tiggle.domain.tag.model.Tag;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
public class TagDto {
    long id;
    String name;

    public static TagDto fromEntity(Tag tag) {
        TagDto dto = new TagDto();
        dto.id = tag.getId();
        dto.name = tag.getName();

        return dto;
    }
}

File: backend/tiggle-root/tiggle/src/main/kotlin/com/side/tiggle/domain/tag/dto/TagDto.kt
package com.side.tiggle.domain.tag.dto

import com.fasterxml.jackson.annotation.JsonCreator
import com.side.tiggle.domain.tag.model.Tag

data class TagDto (
    val id: Long,
    val name: String
) {
    companion object {
        fun fromEntity(tag: Tag): TagDto {
            return TagDto(
                id = tag.id,
                name = tag.name
            )
        }
    }
}